pycbc/io/live.py

```diff
@@ -318,15 +318,15 @@ def save(self, fname):
             with open(self.multipa_file, 'w') as multipaf:
                 json.dump(self.astro_probs, multipaf)
             logging.info('Multi p_astro file saved as %s', self.multipa_file)
-            # Don't save any other files!
-            return
 
         # Save source probabilities in a json file
         if self.probabilities is not None:
             self.prob_file = os.path.join(save_dir, 'src_probs.json')
             with open(self.prob_file, 'w') as probf:
                 json.dump(self.probabilities, probf)
             logging.info('Source probabilities file saved as %s', self.prob_file)
+            # Don't save any other files!
+            return
```

Comment:
I am not clear why you are moving the return from line 322 to here. Could it not be simply removed?

Reply:
Because the subsequent code that creates a pastro/pterr only file should only be carried out in the case that there is no source probability calculation and no multi-component pastro.  This is the same behaviour as before except that we are now creating the pie plot and associated json file as you requested.

Reply:
I.e. I think we still prefer not to create an extra .json file that is fully redundant with other ones and does not correspond to a plot

Reply:
Ok.